pkg/scheduler/internal/queue/scheduling_queue.go

```diff
@@ -822,7 +821,6 @@ func (p *PriorityQueue) AddUnschedulableIfNotPresent(logger klog.Logger, pInfo *
 		p.cond.Broadcast()
 	}
 
-	p.addNominatedPodUnlocked(logger, pInfo.PodInfo, nil)
```

Comment:
I don't understand why it's needed because we handle nominated Pod stuff in handleSchedulingFailure side later.
https://github.com/kubernetes/kubernetes/blob/f2c8d8b4a5ce1ab612fb25f6119ea1ab607af48a/pkg/scheduler/schedule_one.go#L1098

Reply:
Adding the nominated info in the failure handler was added in this commit: https://github.com/kubernetes/kubernetes/blame/27caa6e7271fc14c7e45270e75835be0bd5f6aa1/pkg/scheduler/scheduler.go#L369

If we look at the tree at that point, it already had this line https://github.com/ahg-g/kubernetes/blob/27caa6e7271fc14c7e45270e75835be0bd5f6aa1/pkg/scheduler/internal/queue/scheduling_queue.go#L326

So we likely just forgot to remove it.